ibft/README.md

```diff
@@ -0,0 +1,87 @@
+The IBFT ensures a single and deterministic defined order of transactions in the distributed digital ledger.
+
+There are different working algorithms:
+
+- **Proof Of Work (PoW):**  the first algorithm to build a decentralized ledger. It is an expensive algorithm that requires
+  massive computing efforts and energy consumption from a hardware - the validator node - to deter malicious uses of
+  computing power such as spammers or massive service attacks.
+
+- **Proof of Stake (PoS):**  considered as an enhanced algorithm over the PoW, the Proof of Stake optimize significantly the hardware and power consumption. Blocks are created via various combination of random selection and wealth or age (i.e. stake). The PoS attributes mining power to the proportion of coins held by a miner. This way, a node is limited to mine a percentage of transactions proportionally to his/her stake. Whereas in PoW-based systems an amount of cryptocurrency is created as rewards for miners, the PoS systems usually uses transaction fees as a reward - i.e. Gas fees in Ethereum network.
+
+- **Proof of Authority (PoA):**  known possibly as the best solution, PoA is a consensus mechanism in which a limited number of entities - validators - are allowed to validate transactions. This vetting is based upon someone's identity as a node owner and his/her wealth stake - making malicious behaviour very damaging to their reputation and often leading to penalties. Unlike the PoW mechanism, here there is no competition between validators. The consensus requires almost no computing power, and therefore almost no electricity for its operation.
+
+- **Byzantine Fault Tolerance (BFT):** conceptually, the BFT is the characteristic which defines a system that tolerates the class of failures that belong to the [Byzantine General's Problem](https://medium.com/loom-network/understanding-blockchain-fundamentals-part-1-byzantine-fault-tolerance-245f46fe8419). It is considered as the most difficult class of *failure modes*. In simple words, the algorithm guarantees that the number of traitors do not exceed  one third of the generals. In the absence of BFT< a peer is able to transmit and post false transactions effectively nullifying the blockchain's reliability. Worst than that, given the decentralized nature of the blockchain, no central authority could take it over and repair the damage.
+- ---
+
+## Istanbul Byzantine Fault Tolerance (IBFT)
+
+The implementation of BFT. IBFT is a consensus mechanism that ensures a single, agreed-upon ordering of transactions in the blockchain.
+
+IBFT uses a pool of validation nodes - _Validators_ - operating on an Ethereum network to determine if a proposed block is suitable for addition to the chain.
+
+At a very high level point-of-view we can express the main flow as follows:
+
+1. One validator node is selected randomly by the algorithm as the _Proposer_ and is responsible for computing and constructing a block.
```

Comment:
iBFT Is a 3 phase protocol.
1) pre-prepare: a leader is selected from the group to propose a value to agree upon.
2) upon receiving a valid pre-prepare, other validators broadcast prepare msg
3) upon receiving 2/3 prepare votes, validators broadcast a commit msg.

Reply:
To be added to next commit

Reply:
Done, thanks.